integrations/Braze/browser.js

```diff
@@ -225,23 +226,39 @@ class Braze {
     const { userId } = rudderElement.message;
     const eventName = rudderElement.message.event;
     let { properties } = rudderElement.message;
-
-    window.braze.changeUser(userId);
-    if (eventName && eventName.toLowerCase() === "order completed") {
-      this.handlePurchase(properties, userId);
-    } else {
-      properties = this.handleReservedProperties(properties);
-      window.braze.logCustomEvent(eventName, properties);
+    let canSendCustomEvent = false;
+    if (userId) {
+      window.braze.changeUser(userId);
+      canSendCustomEvent = true;
+    } else if (this.trackAnonymousUser) {
+      window.braze.changeUser(rudderElement.message.anonymousId);
+      canSendCustomEvent = true;
+    }
+    if (eventName && canSendCustomEvent) {
+      if (eventName.toLowerCase() === "order completed") {
+        this.handlePurchase(properties, userId);
+      } else {
+        properties = this.handleReservedProperties(properties);
+        window.braze.logCustomEvent(eventName, properties);
+      }
```

Comment:
What happens if we do not create the user using `anonymousId` I mean not making the `window.braze.changeUser` call and making the `window.braze.logCustomEvent` call only. Does the event show up in the dashboard?

Reply:
events should be assigned to a user